### PR TITLE
Type handling and related exceptions

### PR DESCRIPTION
**Pull request by axunonb**
### YAXLib
* Added `YAXEnumAliasException`
* `YAXLibException` is now an abstract base class in favor of more specific exceptions
* Replaced usage of `YAXLibException` with `YAXObjectTypeMismatch` in `MemberWrapper`and `UdtWrapper`
* Fix for `EnumWrapper`: Did not detect adding duplicate aliases for `Enum`s. Now throws `YAXEnumAliasException` instead of `YAXLibException` when this happens.

### Unit Tests
* TargetFramework net6.0 
* Update to latest dependencies

#### Custom Serializer
* Added test for cases where interface implementation is missing (should throw)

#### Known Types
* Added missing test for known type `Type` (`TypeKnownType`)
* Added missing test for duplicate Enum aliases: should throw
* Added missing tests for `TimeSpan` serialization

### DemoApplication
* TargetFramework net6.0

### Removed unused files
* packages/repositories.config
* YAXLib.csproj.Dotsettings

### CODE REVIEW

**Comment by 304NotModified** [comment]
I dont think I have time to review this before 11 April. Feel free to remind me after that date.

**Comment by axunonb** [comment]
> I dont think I have time to review this before 11 April. Feel free to remind me after that date.

... and I'll be OOO till 28th

File: YAXLib/Exceptions/YAXException.cs [hunk 1]
```diff
@@ -8,7 +8,7 @@ namespace YAXLib.Exceptions
     /// <summary>
     ///     The base for all exception classes of YAXLib
     /// </summary>
-    public class YAXException : Exception
+    public abstract class YAXException : Exception
```
Comment (304NotModified) [inline YAXLib/Exceptions/YAXException.cs:11]: This is a breaking change, but maybe unlikely it's used.

I guess it's good practice to make the contructors now protected

**Review by 304NotModified — APPROVED** [review]

File: YAXLib/Exceptions/YAXException.cs (hunk 1 above)
Comment (304NotModified) [reply YAXLib/Exceptions/YAXException.cs:11]: Doubting if we should fix this

Comment (axunonb) [reply YAXLib/Exceptions/YAXException.cs:11]: Would also qualify as unlikely to be used. Helps avoiding usage of a general exception class by contributors.
